src/go/rpk/pkg/cli/cmd/cluster/offsets.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+package cluster
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/Shopify/sarama"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/ui"
+)
+
+func NewOffsetsCommand(
+	client func() (sarama.Client, error),
+	admin func(sarama.Client) (sarama.ClusterAdmin, error),
+) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "offsets",
+		Short: "Report cluster offset status",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := client()
+			if err != nil {
+				log.Error("Error initializing cluster client")
+				return err
+			}
+
+			admin, err := admin(client)
+			if err != nil {
+				return fmt.Errorf("Error creating admin client: %v", err)
+			}
+			defer admin.Close() // closes underlying client
+
+			return executeOffsetReport(client, admin)
+		},
+	}
+	return cmd
+}
+
+// ConsumerLagRow represents a (group, topic, partition) 3-tuple for reporting
+// consumer lag and will be tagged with the active consumer if one exists.
+type ConsumerLagRow struct {
+	Group           string
+	Topic           string
+	Partition       int32
+	LatestOffset    int64
+	CommittedOffset int64
+	MemberId        string
+}
+
+// executeOffsetReport calculates consumer lag for all consumer groups and
+// reports the summary in table format.
+func executeOffsetReport(
+	client sarama.Client, admin sarama.ClusterAdmin,
+) error {
+	// get all groups and their current members
+	groups, err := describeConsumerGroups(admin)
+	if err != nil {
+		return fmt.Errorf("Error retrieving group information: %v", err)
+	}
+
+	// collect consumer lag information for all groups/members
+	var consumers []ConsumerLagRow
+	for _, groupDesc := range groups {
+		c, err := getConsumerLag(client, admin, groupDesc)
+		if err != nil {
+			return fmt.Errorf("Error listing consumer groups: %v", err)
+		}
+		consumers = append(consumers, c...)
```

Comment:
It seems possible that group membership can change during this operation, perhaps `continue` is better than logging the error, at least for a subset of errors.

Reply:
Good idea, fixed.